src/analysis/calculus/fderiv/exp.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import measure_theory.integral.fund_thm_calculus
+import analysis.normed_space.exponential
+import analysis.quaternion
+import algebra.lie.of_associative
+import analysis.special_functions.exponential
+import analysis.calculus.fderiv_symmetric
+
+/-! More lemmas aboutderiviatives of `exp`.
+
+This follows https://physics.stackexchange.com/a/41671/185147. -/
+
+variables {𝔸 𝔹 : Type*}
+variables [normed_ring 𝔸] [normed_algebra ℝ 𝔸] [complete_space 𝔸]
+
+-- to make the goal view readable
+notation (name := deriv) `∂` binders `, ` r:(scoped:67 f, deriv f) := r
+local notation `e` := exp ℝ
+
+lemma has_deriv_at_exp_smul_const (A : 𝔸) (t : ℝ) :
+  has_deriv_at (λ t : ℝ, exp ℝ (t • A)) (A * exp ℝ (t • A)) t := sorry
+
+lemma has_deriv_at_exp_smul_const' (A : 𝔸) (t : ℝ) :
+  has_deriv_at (λ t : ℝ, exp ℝ (t • A)) (exp ℝ (t • A) * A) t :=
+begin
+  convert has_deriv_at_exp_smul_const A t using 1,
+  refine commute.exp_left _ _,
+  refine (commute.refl _).smul_left _,
+end
+
+lemma deriv_exp_aux (A : ℝ → 𝔸) (r t : ℝ)
+  (hA : differentiable_at ℝ A r) :
+  exp ℝ (-t • A r) * deriv (λ x, exp ℝ (t • A x)) r =
+    (∫ s : ℝ in 0..t, exp ℝ (-s • A r) * deriv A r * exp ℝ (s • A r)) :=
+begin
+  revert t,
+  rw ←function.funext_iff,
+  -- two functions agree if their derivatives agree and they agree at zero.
+  do { `(%%lhs = %%rhs) ← tactic.target,
+    let t := ``(%%lhs 0 = %%rhs 0 ∧ deriv %%lhs = deriv %%rhs),
+    tactic.interactive.suffices none (some t)},
+  { sorry },
```

Comment:
This is almost certainly not the right way to do this; and it wouldn't surprise me if this is not even true...

Reply:
The fundamental result for that is definitely [convex.is_const_of_fderiv_within_eq_zero](https://leanprover-community.github.io/mathlib_docs/analysis/calculus/mean_value.html#convex.is_const_of_fderiv_within_eq_zero) (which could be weakened to `star_convex` instead of `convex` IIRC, but you shouldn't need that). That said we should definitely add the corollary for two functions (you apply the this lemma to the difference to get that they are equal).

Reply:
Thanks! I made #19060 with this result.